src/components/BulkOperationDetailsPage/BulkOperationDetails.jsx

```diff
@@ -0,0 +1,196 @@
+import React, { useState, useMemo } from 'react';
+import 'moment-timezone';
+import PropTypes from 'prop-types';
+import Papa from 'papaparse';
+import {
+  Button, DataTable, useToggle, FullscreenModal, Stack,
+} from '@openedx/paragon';
+
+import { formatDateTime } from '../../utils';
+import './styles.scss';
+
+const ImageCell = ({ value }) => (
+  value ? (
+    <a href={value} target="_blank" rel="noopener noreferrer">
+      View Image
+    </a>
+  ) : null
+);
+
+const parseCSV = async (url) => {
+  const response = await fetch(url);
+  const text = await response.text();
+
+  return new Promise((resolve, reject) => {
+    Papa.parse(text, {
+      header: true,
+      skipEmptyLines: true,
+      complete: (results) => {
+        const { data, meta } = results;
+        const headers = meta.fields;
+        resolve({ headers, rows: data });
+      },
+      error: (error) => {
+        reject(error);
+      },
+    });
+  });
+};
+
+const BulkOperationDetails = ({ task }) => {
+  const [csvRows, setCsvRows] = useState([]);
+  const [csvHeaders, setCsvHeaders] = useState([]);
+  const [isOpen, open, close] = useToggle(false);
+  const [csvPreviewError, setCsvPreviewError] = useState(null);
+  const DEFAULT_CSV_PREVIEW_PAGE_SIZE = 10;
+
+  const taskInfoRows = [
+    { key: 'Task ID', value: task.task_id },
+    { key: 'Task type', value: task.task_type },
+    { key: 'Uploaded by', value: task.uploaded_by },
+    { key: 'Status', value: task.status },
+    { key: 'Created', value: formatDateTime(task.created) },
+    { key: 'Modified', value: formatDateTime(task.modified) },
+  ];
+
+  const taskInfoColumns = [
+    { Header: 'Field', accessor: 'key' },
+    { Header: 'Value', accessor: 'value' },
+  ];
+
+  const handlePreviewCSV = async () => {
+    try {
+      const { headers, rows } = await parseCSV(task.csv_file);
+      setCsvHeaders(headers);
+      setCsvRows(rows);
+    } catch (error) {
+      console.error('Failed to preview CSV:', error);
+      setCsvPreviewError(`Failed to preview CSV: ${error.message}`);
+    }
+  };
+
+  const getDataTableColumns = (headers) => headers.map((header) => {
+    const isImageColumn = header.toLowerCase() === 'image';
+
+    const column = {
+      Header: header,
+      accessor: header,
+    };
+
+    if (isImageColumn) {
+      column.Cell = ImageCell;
+    }
+
+    return column;
+  });
+
+  const columns = useMemo(() => { // eslint-disable-line react-hooks/rules-of-hooks
+    if (!csvHeaders || csvHeaders.length === 0) {
+      return [];
+    }
+    return getDataTableColumns(csvHeaders);
+  }, [csvHeaders]);
+
+  const data = useMemo(() => { // eslint-disable-line react-hooks/rules-of-hooks
+    if (!csvHeaders || csvHeaders.length === 0 || !csvRows || csvRows.length === 0) {
+      return [];
+    }
+    return csvRows;
+  }, [csvRows, csvHeaders]);
```

Comment:
why is useMemo checking on both rows and headers?

Reply:
Shouldn't we refresh data, either of them changes?
I think there’s a possibility that the CSV might contain only headers, just thinking it as a case.

Reply:
Then should this not be done for headers useMemo too?

Reply:
hmm yes i believe.